### PR TITLE
feat: new module network/vpn-gateway

### PR DESCRIPTION
**Pull request by fabmas**
## Description

New network/virtual-network-gateway module migrated from CARML to AVM

[avm.res.network.vpn-gateway](https://github.com/fabmas/bicep-registry-modules/actions/workflows/avm.res.network.vpn-gateway.yml)

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -110,7 +110,7 @@
 /avm/res/network/virtual-network/ @Azure/avm-res-network-virtualnetwork-module-owners-bicep @Azure/avm-core-team-technical-bicep
 #/avm/res/network/virtual-network-gateway/ @Azure/avm-res-network-virtualnetworkgateway-module-owners-bicep @Azure/avm-core-team-technical-bicep
 #/avm/res/network/virtual-wan/ @Azure/avm-res-network-virtualwan-module-owners-bicep @Azure/avm-core-team-technical-bicep
-#/avm/res/network/vpn-gateway/ @Azure/avm-res-network-vpngateway-module-owners-bicep @Azure/avm-core-team-technical-bicep
+/avm/res/network/vpn-gateway/ @Azure/avm-res-network-vpngateway-module-owners-bicep @Azure/avm-core-team-technical-bicep
```
Comment (eriqua) [inline .github/CODEOWNERS:113]: Same here. All module owner teams should be member of parent team `avm-technical-reviewers-bicep` before the PR approval

Comment (eriqua) [reply .github/CODEOWNERS:113]: Solved

File: avm/res/network/vpn-gateway/main.bicep [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  name: '${deployment().name}-NATRule-${index}'
+  params: {
+    name: natRule.name
+    vpnGatewayName: vpnGateway.name
+    externalMappings: contains(natRule, 'externalMappings') ? natRule.externalMappings : []
+    internalMappings: contains(natRule, 'internalMappings') ? natRule.internalMappings : []
+    ipConfigurationId: contains(natRule, 'ipConfigurationId') ? natRule.ipConfigurationId : ''
+    mode: contains(natRule, 'mode') ? natRule.mode : ''
+    type: contains(natRule, 'type') ? natRule.type : ''
+    enableDefaultTelemetry: enableReferencedModulesTelemetry
```
Comment (AlexanderSehr) [inline avm/res/network/vpn-gateway/main.bicep:116]:
```suggestion
```

File: avm/res/network/vpn-gateway/main.bicep [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+@description('Optional. Tags of the resource.')
+param tags object?
+
+@description('Optional. The lock settings of the service.')
+param lock lockType
+
+@description('Optional. Enable telemetry via a Globally Unique Identifier (GUID).')
+param enableTelemetry bool = true
+
+var enableReferencedModulesTelemetry = false
```
Comment (AlexanderSehr) [inline avm/res/network/vpn-gateway/main.bicep:41]:
```suggestion
```

File: avm/res/network/vpn-gateway/nat-rule/main.bicep [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  name: 'pid-47ed15a6-730a-4827-bcb4-0fd963ffbd82-${uniqueString(deployment().name)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+    }
+  }
+}
```
Comment (AlexanderSehr) [inline avm/res/network/vpn-gateway/nat-rule/main.bicep:49]:
```suggestion
```

File: avm/res/network/vpn-gateway/vpn-connection/main.bicep [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  name: 'pid-47ed15a6-730a-4827-bcb4-0fd963ffbd82-${uniqueString(deployment().name)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+    }
+  }
+}
```
Comment (AlexanderSehr) [inline avm/res/network/vpn-gateway/vpn-connection/main.bicep:71]:
```suggestion
```

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]
Remember to also run the `Set-AVMModule` utility after applying all changes. Ideally, also re-run the pipeline.

**Review by AlexanderSehr — APPROVED** [review]

**Review by AlexanderSehr — CHANGES_REQUESTED** [review]
Hey @fabmas , the PR looks good to me. However, please note that there is one small conflict that needs resolving first.

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**